### PR TITLE
Handling packets recv on unknown channels

### PR DESCRIPTION
**Pull request by mpoke**
Closes https://github.com/cosmos/interchain-security/issues/292

### CODE REVIEW

File: x/ccv/consumer/keeper/relay.go [hunk 1]
```diff
@@ -25,8 +25,10 @@ func (k Keeper) OnRecvVSCPacket(ctx sdk.Context, packet channeltypes.Packet, new
 	// get the provider channel
 	providerChannel, found := k.GetProviderChannel(ctx)
 	if found && providerChannel != packet.DestinationChannel {
-		// VSC packet was sent on a channel different than the provider channel
-		return utils.OnRecvPacketOnUnknownChannel(ctx, k.scopedKeeper, k.channelKeeper, packet)
+		// VSC packet was sent on a channel different than the provider channel;
+		// this should never happen
+		panic(fmt.Errorf("VSCPacket received on unknown channel %s; expected: %s",
+			packet.DestinationChannel, providerChannel))
```
Comment (danwt) [inline x/ccv/consumer/keeper/relay.go:31]: I'm not sure I understand why this is a panic. From reading https://github.com/cosmos/interchain-security/issues/292 I understand that the consumer can receive packets from channels which arent OPEN on the provider. It sounds like recoverable behavior, not panic behavior.

Comment (danwt) [inline x/ccv/consumer/keeper/relay.go:31]: I'm not sure I understand why this is a panic. From reading https://github.com/cosmos/interchain-security/issues/292 I understand that the consumer can receive packets from channels which arent OPEN on the provider. It sounds like recoverable behavior, not panic behavior.

File: x/ccv/provider/keeper/relay.go [hunk 2]
```diff
@@ -87,10 +88,8 @@ func (k Keeper) CompleteMaturedUnbondingOps(ctx sdk.Context) {
 // OnAcknowledgementPacket handles acknowledgments for sent VSC packets
 func (k Keeper) OnAcknowledgementPacket(ctx sdk.Context, packet channeltypes.Packet, ack channeltypes.Acknowledgement) error {
 	if err := ack.GetError(); err != "" {
-		// Either the VSC packet data could not be successfully decoded
-		// or the VSC packet was sent on a channel other than the established
-		// provider channel and ChanCloseInit failed.
-		// Neither of these should ever happen.
+		// The VSC packet data could not be successfully decoded.
+		// This should never happen.
```
Comment (danwt) [inline x/ccv/provider/keeper/relay.go:92]: I think it would be more precise to say that it should rarely happen.

**Review by danwt — APPROVED** [review]

File: x/ccv/consumer/keeper/relay.go (hunk 1 above)
Comment (mpoke) [reply x/ccv/consumer/keeper/relay.go:31]: See https://github.com/cosmos/interchain-security/issues/292#issuecomment-1231334565

File: x/ccv/provider/keeper/relay.go (hunk 2 above)
Comment (mpoke) [reply x/ccv/provider/keeper/relay.go:92]: This is a programming error. Should never happen.

File: x/ccv/consumer/keeper/relay.go (hunk 1 above)
Comment (danwt) [reply x/ccv/consumer/keeper/relay.go:31]: I'm not sure if we should have this branch at all. Having it implies it can happen sometimes.

Comment (jtremback) [reply x/ccv/consumer/keeper/relay.go:31]: Agree with Dan. Unless it can somehow happen, but it sounds like that would require an incorrect provider.

**Review by jtremback — APPROVED** [review]
Nice! Love to see stuff getting deleted

**Pull request merged**